### PR TITLE
[JBPM-9545] add indexes to rhpam to improve the performance

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9545

### CODE REVIEW

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql [hunk 1]
```diff
@@ -1,818 +1,191 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create table Attachment (id bigint generated by default as identity, accessType integer, attachedAt timestamp, attachmentContentId bigint not null, contentType varchar(255), name varchar(255), attachment_size integer, attachedBy_id varchar(255), TaskData_Attachments_Id bigint, primary key (id));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql:1]: I think it should be good to keep the format as we already have as it makes easier to read - what do you think?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-drop-schema.sql [hunk 2]
```diff
@@ -1,45 +1,90 @@
...
+alter table PeopleAssignments_Stakeholders drop constraint FK9uy76cu650rg1nnkrtjwj1e9t;
+alter table PeopleAssignments_Stakeholders drop constraint FKaeyk4nwslvx0jywjomjq7083i;
+alter table Reassignment drop constraint FKessy30safh44b30f1cfoujv2k;
+alter table Reassignment_potentialOwners drop constraint FKsqrmpvehlc4qe9i0km22nmkjl;
+alter table Reassignment_potentialOwners drop constraint FKftegfexshix752bh2jfxf6bnh;
+alter table Task drop constraint FK48d1bfgwf0jqow1yk8ku4xcpi;
+alter table Task drop constraint FKpmkxvqq63aed2y2boruu53a0s;
+alter table Task drop constraint FKexuboqnbla7jfyyesyo61ucmb;
+alter table task_comment drop constraint FKqb4mkarf209y9546w7n75lb7a;
+alter table task_comment drop constraint FKm2mwc1ukcpdsiqwgkoroy6ise;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-drop-schema.sql:45]: why these all alter table now being added to this script?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql [hunk 3]
```diff
@@ -1,819 +1,191 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create table Attachment (id bigint generated by default as identity, accessType integer, attachedAt timestamp, attachmentContentId bigint not null, contentType varchar(255), name varchar(255), attachment_size integer, attachedBy_id varchar(255), TaskData_Attachments_Id bigint, primary key (id));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql:1]: Same as db2 - format is easier to read as it is already

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/h2/h2-jbpm-schema.sql [hunk 4]
```diff
@@ -1,821 +1,191 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create table Attachment (id bigint generated by default as identity, accessType integer, attachedAt timestamp, attachmentContentId bigint not null, contentType varchar(255), name varchar(255), attachment_size integer, attachedBy_id varchar(255), TaskData_Attachments_Id bigint, primary key (id))
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/h2/h2-jbpm-schema.sql:1]: same format thing

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/hsqldb/hsqldb-jbpm-drop-schema.sql [hunk 5]
```diff
@@ -1,3 +1,48 @@
+alter table Attachment drop constraint FKd5xpm81gxg8n40167lbu5rbfb;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/hsqldb/hsqldb-jbpm-drop-schema.sql:1]: same question regarding alter table commands

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-drop-schema.sql [hunk 6]
```diff
@@ -1,47 +1,3 @@
...
-alter table Notification_email_header drop foreign key FK_eth4nvxn21fk1vnju85vkjrai;
-alter table Notification_Recipients drop foreign key FK_blf9jsrumtrthdaqnpwxt25eu;
-alter table Notification_Recipients drop foreign key FK_3l244pj8sh78vtn9imaymrg47;
-alter table PeopleAssignments_BAs drop foreign key FK_t38xbkrq6cppifnxequhvjsl2;
-alter table PeopleAssignments_BAs drop foreign key FK_omjg5qh7uv8e9bolbaq7hv6oh;
-alter table PeopleAssignments_ExclOwners drop foreign key FK_pth28a73rj6bxtlfc69kmqo0a;
-alter table PeopleAssignments_ExclOwners drop foreign key FK_b8owuxfrdng050ugpk0pdowa7;
-alter table PeopleAssignments_PotOwners drop foreign key FK_tee3ftir7xs6eo3fdvi3xw026;
-alter table PeopleAssignments_PotOwners drop foreign key FK_4dv2oji7pr35ru0w45trix02x;
-alter table PeopleAssignments_Recipients drop foreign key FK_4g7y3wx6gnokf6vycgpxs83d6;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-drop-schema.sql:33]: why are we now removing these alter table?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql [hunk 7]
```diff
@@ -1,841 +1,191 @@
...
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
-
-    create index IDX_PAsPot_TaskEntity ON PeopleAssignments_PotOwners (task_id, entity_id);
+create table Attachment (id bigint not null auto_increment, accessType integer, attachedAt datetime, attachmentContentId bigint not null, contentType varchar(255), name varchar(255), attachment_size integer, attachedBy_id varchar(255), TaskData_Attachments_Id bigint, primary key (id)) engine=MyISAM;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql:1]: same format thing

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/oracle/oracle-springboot-jbpm-schema.sql [hunk 8]
```diff
@@ -72,7 +72,7 @@ create table SessionInfo (id number(19,0) not null, lastModificationDate timesta
...
 create table task_comment (id number(19,0) not null, addedAt timestamp, text clob, addedBy_id varchar2(255 char), TaskData_Comments_Id number(19,0), primary key (id));
 create table TaskDef (id number(19,0) not null, name varchar2(255 char), priority number(10,0) not null, primary key (id));
-create table TaskEvent (id number(19,0) not null, logTime timestamp, message varchar2(255 char), processInstanceId number(19,0), taskId number(19,0), type varchar2(255 char), userId varchar2(255 char), OPTLOCK number(10,0), workItemId number(19,0), correlationKey varchar(255), processType number(1,0), primary key (id));
+create table TaskEvent (id number(19,0) not null, correlationKey varchar2(255 char), logTime timestamp, message varchar2(255 char), processInstanceId number(19,0), processType number(10,0), taskId number(19,0), type varchar2(255 char), userId varchar2(255 char), OPTLOCK number(10,0), workItemId number(19,0), primary key (id));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/oracle/oracle-springboot-jbpm-schema.sql:75]: processType is being changed from `number(1,0)` to `number(10,0)` - is this a typo?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-bytea-jbpm-schema.sql [hunk 9]
```diff
@@ -40,7 +40,7 @@ create table CaseIdInfo (id int8 not null, caseIdPrefix varchar(255), currentVal
 create table CaseRoleAssignmentLog (id int8 not null, caseId varchar(255), entityId varchar(255), processInstanceId int8 not null, roleName varchar(255), type int4 not null, primary key (id));
 create table Content (id int8 not null, content bytea, primary key (id));
 create table ContextMappingInfo (mappingId int8 not null, CONTEXT_ID varchar(255) not null, KSESSION_ID int8 not null, OWNER_ID varchar(255), OPTLOCK int4, primary key (mappingId));
-create table CorrelationKeyInfo (keyId int8 not null, name varchar(255) not null, processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
+create table CorrelationKeyInfo (keyId int8 not null, name varchar(255), processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-bytea-jbpm-schema.sql:43]: Why this change? `name` field should be `not null`

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-jbpm-drop-schema.sql [hunk 10]
```diff
@@ -1,48 +1,48 @@
...
+alter table BooleanExpression drop constraint FKqth56a8k6d8pv6ngsu2vjp4kj;
+alter table CorrelationPropertyInfo drop constraint FKbchyl7kb8i6ghvi3dbr86bgo0;
+alter table Deadline drop constraint FK361ggw230po88svgfasg36i2w;
+alter table Deadline drop constraint FKpeiadnoy228t35213t63c3imm;
+alter table Delegation_delegates drop constraint FKewkdyi0wrgy9byp6abyglpcxq;
+alter table Delegation_delegates drop constraint FK85x3trafk3wfbrv719cafr591;
+alter table ErrorInfo drop constraint FKdarp6ushq06q39jmij3fdpdbs;
+alter table Escalation drop constraint FK37v8ova8ti6jiblda7n6j298e;
+alter table EventTypes drop constraint FKj0o3uve2nqo5yrjwrkc9jfttq;
+alter table I18NText drop constraint FKcd6eb4q62d9ab8p0di8pb99ch;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-jbpm-drop-schema.sql:12]: why these all changes?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-springboot-bytea-jbpm-schema.sql [hunk 11]
```diff
@@ -40,7 +40,7 @@ create table CaseIdInfo (id int8 not null, caseIdPrefix varchar(255), currentVal
 create table CaseRoleAssignmentLog (id int8 not null, caseId varchar(255), entityId varchar(255), processInstanceId int8 not null, roleName varchar(255), type int4 not null, primary key (id));
 create table Content (id int8 not null, content bytea, primary key (id));
 create table ContextMappingInfo (mappingId int8 not null, CONTEXT_ID varchar(255) not null, KSESSION_ID int8 not null, OWNER_ID varchar(255), OPTLOCK int4, primary key (mappingId));
-create table CorrelationKeyInfo (keyId int8 not null, name varchar(255) not null, processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
+create table CorrelationKeyInfo (keyId int8 not null, name varchar(255), processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-springboot-bytea-jbpm-schema.sql:43]: same - `name` should be `not null`

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-springboot-jbpm-schema.sql [hunk 12]
```diff
@@ -40,7 +40,7 @@ create table CaseIdInfo (id int8 not null, caseIdPrefix varchar(255), currentVal
 create table CaseRoleAssignmentLog (id int8 not null, caseId varchar(255), entityId varchar(255), processInstanceId int8 not null, roleName varchar(255), type int4 not null, primary key (id));
 create table Content (id int8 not null, content oid, primary key (id));
 create table ContextMappingInfo (mappingId int8 not null, CONTEXT_ID varchar(255) not null, KSESSION_ID int8 not null, OWNER_ID varchar(255), OPTLOCK int4, primary key (mappingId));
-create table CorrelationKeyInfo (keyId int8 not null, name varchar(255) not null, processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
+create table CorrelationKeyInfo (keyId int8 not null, name varchar(255), processInstanceId int8 not null, OPTLOCK int4, primary key (keyId));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-springboot-jbpm-schema.sql:43]: same

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver/sqlserver-jbpm-drop-schema.sql [hunk 13]
```diff
@@ -1,48 +1,48 @@
...
-alter table Task drop constraint FK_k02og0u71obf1uxgcdjx9rcjc;
-alter table task_comment drop constraint FK_aax378yjnsmw9kb9vsu994jjv;
-alter table task_comment drop constraint FK_1ws9jdmhtey6mxu7jb0r0ufvs;
+alter table Attachment drop constraint FKd5xpm81gxg8n40167lbu5rbfb;
+alter table Attachment drop constraint FKjj9psk52ifamilliyo16epwpc;
+alter table BooleanExpression drop constraint FKqth56a8k6d8pv6ngsu2vjp4kj;
+alter table CorrelationPropertyInfo drop constraint FKbchyl7kb8i6ghvi3dbr86bgo0;
+alter table Deadline drop constraint FK361ggw230po88svgfasg36i2w;
+alter table Deadline drop constraint FKpeiadnoy228t35213t63c3imm;
+alter table Delegation_delegates drop constraint FKewkdyi0wrgy9byp6abyglpcxq;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver/sqlserver-jbpm-drop-schema.sql:7]: same

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver/sqlserver-jbpm-schema.sql [hunk 14]
```diff
@@ -1,822 +1,191 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create table Attachment (id numeric(19,0) identity not null, accessType int, attachedAt datetime, attachmentContentId numeric(19,0) not null, contentType varchar(255), name varchar(255), attachment_size int, attachedBy_id varchar(255), TaskData_Attachments_Id numeric(19,0), primary key (id));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver/sqlserver-jbpm-schema.sql:1]: same related to format issue

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver2008/sqlserver2008-jbpm-drop-schema.sql [hunk 15]
```diff
@@ -1,48 +1,48 @@
...
-alter table task_comment drop constraint FK_aax378yjnsmw9kb9vsu994jjv;
-alter table task_comment drop constraint FK_1ws9jdmhtey6mxu7jb0r0ufvs;
+alter table Attachment drop constraint FKd5xpm81gxg8n40167lbu5rbfb;
+alter table Attachment drop constraint FKjj9psk52ifamilliyo16epwpc;
+alter table BooleanExpression drop constraint FKqth56a8k6d8pv6ngsu2vjp4kj;
+alter table CorrelationPropertyInfo drop constraint FKbchyl7kb8i6ghvi3dbr86bgo0;
+alter table Deadline drop constraint FK361ggw230po88svgfasg36i2w;
+alter table Deadline drop constraint FKpeiadnoy228t35213t63c3imm;
+alter table Delegation_delegates drop constraint FKewkdyi0wrgy9byp6abyglpcxq;
+alter table Delegation_delegates drop constraint FK85x3trafk3wfbrv719cafr591;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver2008/sqlserver2008-jbpm-drop-schema.sql:8]: same

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver2008/sqlserver2008-jbpm-schema.sql [hunk 16]
```diff
@@ -1,822 +1,191 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create table Attachment (id bigint identity not null, accessType int, attachedAt datetime2, attachmentContentId bigint not null, contentType varchar(255), name varchar(255), attachment_size int, attachedBy_id varchar(255), TaskData_Attachments_Id bigint, primary key (id));
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sqlserver2008/sqlserver2008-jbpm-schema.sql:1]: same related to format thing

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sybase/sybase-jbpm-drop-schema.sql [hunk 17]
```diff
@@ -1,92 +1,92 @@
-alter table Attachment drop constraint FK1C93543D937BFB5
+alter table Attachment drop constraint FKd5xpm81gxg8n40167lbu5rbfb
     go
-alter table Attachment drop constraint FK1C9354333CA892A
+alter table Attachment drop constraint FKjj9psk52ifamilliyo16epwpc
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sybase/sybase-jbpm-drop-schema.sql:3]: probably all these changes are because of the reformat change. I think we should keep them as it is

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/sybase/sybase-jbpm-schema.sql [hunk 18]
```diff
@@ -1,1011 +1,382 @@
...
-        add constraint FK482F79D572B3A123
-        foreign key (entity_id)
-        references OrganizationalEntity
-    go
-
-    alter table PeopleAssignments_Stakeholders
-        add constraint FK482F79D5786553A5
-        foreign key (task_id)
-        references Task
+create table Attachment (id numeric(19,0) identity not null, accessType int null, attachedAt datetime null, attachmentContentId numeric(19,0) not null, contentType varchar(255) null, name varchar(255) null, attachment_size int null, attachedBy_id varchar(255) null, TaskData_Attachments_Id numeric(19,0) null, primary key (id))
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/sybase/sybase-jbpm-schema.sql:1]: same for the format

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-jbpm-schema.sql [hunk 19]
```diff
@@ -1,888 +1,224 @@
...
-    create index IDX_AuditTaskImpl_workItemId on AuditTaskImpl(workItemId);
-    create index IDX_AuditTaskImpl_name on AuditTaskImpl(name);
-    create index IDX_AuditTaskImpl_processId on AuditTaskImpl(processId);
-    create index IDX_AuditTaskImpl_status on AuditTaskImpl(status);
-
-    create index IDX_TaskVariableImpl_taskId on TaskVariableImpl(taskId);
-    create index IDX_TaskVariableImpl_pInstId on TaskVariableImpl(processInstanceId);
-    create index IDX_TaskVariableImpl_processId on TaskVariableImpl(processId);
\ No newline at end of file
+create sequence ATTACHMENT_ID_SEQ start 1 increment 1;
```
Comment (afalhambra) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/postgresql/postgresql-jbpm-schema.sql:1]: same for the format thing

**Review by afalhambra — CHANGES_REQUESTED** [review]
There quite a lot changes which I think they are not really related to including these indexes. 

Mainly:
- There is a new format of the create sql scripts which are drastically changed, making them a little less readable.
- As part for this change, the drop scripts have changed due to the new constraints, and some alter tables commands are newly added where they were not there before, or some have been removed.
- A couple of changes in a `name` field for the CorrelationPropertyInfo table are changes to nullable. It should be not null as it is already.

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Hashes shouldn't be modified as this creates a backward incompatibility with previous release scripts.
Missing changes at upgrade scripts.

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
# There is a new format of the create sql scripts which are drastically changed, making them a little less readable. > The exercise tried here was to update to reflect the exact schema is provided in the JPA pojos.
# Hashes are not something that matters in a db from scratch
# As part for this change, the drop scripts have changed due to the new constraints, and some alter tables commands are newly added where they were not there before, or some have been removed. -> they were autogenerated by hibernate.

File: jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java [hunk 20]
```diff
@@ -35,7 +35,8 @@
 @Entity
 @Table(name = "NodeInstanceLog", indexes = {@Index(name = "IDX_NInstLog_pInstId", columnList = "processInstanceId"),
                                         @Index(name = "IDX_NInstLog_nodeType", columnList = "nodeType"),
-                                        @Index(name = "IDX_NInstLog_pId", columnList = "processId")})
+                                        @Index(name = "IDX_NInstLog_pId", columnList = "processId"),
+                                        @Index(name = "IDX_NInstLog_workItemId", columnList = "workItemId")})
```
Comment (gmunozfe) [inline jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java:39]: There's a discrepancy in naming: here it's "IDX_NInstLog...", but in most of the files (all but h2) is "NodeInstanceLog...", please unify them

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Index name is not matching with the Entity annotation in most of the files but h2

File: jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java (hunk 20 above)
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java:39]: onde

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql [hunk 21]
```diff
@@ -803,6 +805,7 @@
     create index IDX_NInstLog_pInstId on NodeInstanceLog(processInstanceId);
     create index IDX_NInstLog_nodeType on NodeInstanceLog(nodeType);
     create index IDX_NInstLog_pId on NodeInstanceLog(processId);
+    create index IDX_NInstLog_pId on NodeInstanceLog(workItemId);
```
Comment (gmunozfe) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql:808]: typo, it should be IDX_NInstLog_workItemId

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql [hunk 22]
```diff
@@ -802,6 +804,7 @@
     create index IDX_NInstLog_pInstId on NodeInstanceLog(processInstanceId);
     create index IDX_NInstLog_nodeType on NodeInstanceLog(nodeType);
     create index IDX_NInstLog_pId on NodeInstanceLog(processId);
+	create index IDX_NInstLog_pId on NodeInstanceLog(workItemId);
```
Comment (gmunozfe) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql:807]: typo, it should be IDX_NInstLog_workItemId

**Review by gmunozfe — CHANGES_REQUESTED** [review]
still 2 typos

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql (hunk 22 above)
Comment (elguardian) [reply jbpm-db-scripts/src/main/resources/db/ddl-scripts/db2/db2-jbpm-schema.sql]: ok

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql (hunk 21 above)
Comment (elguardian) [reply jbpm-db-scripts/src/main/resources/db/ddl-scripts/derby/derby-jbpm-schema.sql]: ok

**Review by gmunozfe — APPROVED** [review]
Looks good to me, excellent job @elguardian !
